### PR TITLE
Begins setup of system ID pipeline, starting with the swerve module

### PR DESCRIPTION
**Pull request by jkleiber**
**Changes**
- SysID pipeline implemented, can be extended for other subsystems
- Swerve module sysID started, results pending

**Verification**
- System ID plots:
Screenshot from 2023-02-08 22-30-26
- With Kp = 1.5, Ki = 0.01, and Kd = 0.1, I got the control graph:
Screenshot from 2023-02-08 23-18-18

### CODE REVIEW

File: robot_modeling/common/linear_systems.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    # Find u(t) at using interpolation
+    t0 = t[0]
+    t = t - t0
+    u_t = interp1d(t, ctrl)
+
+    # Find the end time for simulation
+    tf = np.max(t)
+
+    # Simulate an arbitrary linear system given system ID parameters.
+    x_traj = solve_ivp(linear_sys, t_span=[0, tf], y0=x0, t_eval=t, args=(u_t, params))
```
Comment (danielbrownmsm) [inline robot_modeling/common/linear_systems.py:117]: I remember this being bad, but if it works it works

File: robot_modeling/common/load_frc_logs.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        if line.strip() == "":
+            continue
+        try:
+            timestamp, id, data = line.split(",")
+        except Exception:
+            print(line)
+            continue
+        
+        try:
+            if id in set(sysid_data.keys()):
```
Comment (danielbrownmsm) [inline robot_modeling/common/load_frc_logs.py:18]: that's a much better way to do that, we should open a PR for the wpilib_process_logs.py file to be better. Actually, I guess I can just open a PR for feat/better-log-processing

File: robot_modeling/swerve/swerve_ctrl.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+from common.control_utils import clamp, clamp_update
+
+# Module name used by main.py as a user argument.
+SWERVE_MODULE_CTRL = "swerve_ctrl"
+
+# Swerve module dynamics
+SWERVE_MODEL = second_order_sys_x2_deadband
+
+# Swerve module constants
+params = {
```
Comment (danielbrownmsm) [inline robot_modeling/swerve/swerve_ctrl.py:20]: what'd you calculate these from?

**Review by danielbrownmsm — APPROVED** [review]
I don't really know what this is doing, but yeah, looks cool

File: robot_modeling/swerve/swerve_ctrl.py (hunk 3 above)
Comment (jkleiber) [reply robot_modeling/swerve/swerve_ctrl.py:20]: I got these from the system ID results

File: robot_modeling/common/load_frc_logs.py (hunk 2 above)
Comment (jkleiber) [reply robot_modeling/common/load_frc_logs.py:18]: for what it's worth: if the new logging framework goes in at some point, we won't need to run a script to process the logs - they'll already be in the CSV format we already use pretty much

File: robot_modeling/common/linear_systems.py (hunk 1 above)
Comment (jkleiber) [reply robot_modeling/common/linear_systems.py:117]: `solve_ivp` is very finnicky about what arguments you give it, but this works much better than last year's setup

File: robot_modeling/swerve/swerve_ctrl.py (hunk 3 above)
Comment (danielbrownmsm) [reply robot_modeling/swerve/swerve_ctrl.py:20]: but the code is used to run the system ID, right?

**Pull request merged**